src/progression.ts

```diff
@@ -14,31 +23,53 @@ type HitDice = {
     n: number
 }
 
-type Armor = string
-type Weapon = string
-type Tool = string
-type SavingThrow = string
-type Skill = string
-type Other = Item | string
-type Proficience = Armor | Weapon | Tool | SavingThrow | Skill | Other
-
 type HpFormula = {
     init: number,
     modifier: string
 }
 
-type Feature = Item | string
-
-type Equipment = Item
-
 type Milestone = {
     level: number,
     proficiencyBonus: number,
-    features: [Feature]
+    /*features: [Feature]*/
+}
+
+type Spellcasting = {
+    spellcastingAbility: boolean,
+    spellcastingType: Item | string,
+    hasSpells: string,

```

Comment:
what would be the value of that field?

Reply:
there. spellcasting refers to class features. it should be grouped. no reason to point as separate type. then 'hasSpells' may be moved to 'classProgression' as additional field with possibly value of string[] or null